File: 0013-roman-to-integer/0013-roman-to-integer.java
import java.util.HashMap;

class Solution {
    public int romanToInt(String s) {
        HashMap<Character, Integer> map = new HashMap<>();
        map.put('I', 1);
        map.put('V', 5);
        map.put('X', 10);
        map.put('L', 50);
        map.put('C', 100);
        map.put('D', 500);
        map.put('M', 1000);

        int integerVal = 0;
        int ind = 0;

        while (ind < s.length()) {
            if (ind < s.length() - 1 && map.get(s.charAt(ind)) < map.get(s.charAt(ind + 1))) {
                integerVal += map.get(s.charAt(ind + 1)) - map.get(s.charAt(ind));
                ind += 2; 
            } else {
                integerVal += map.get(s.charAt(ind));
                ind++; 
            }
        }
        return integerVal;
    }
}

File: 0013-roman-to-integer/0013-roman-to-integer.cs
public class Solution {
    public int RomanToInt(string s) {
        var romans = new Dictionary<char, int> {
            {'I', 1},
            {'V', 5},
            {'X', 10},
            {'L', 50},
            {'C', 100},
            {'D', 500},
            {'M', 1000}
        };
        int integerValue = 0;
        int ind = 0;
        while (ind < s.Length) {
            if (ind < s.Length - 1 && romans[s[ind]] < romans[s[ind + 1]]) {
                integerValue += romans[s[ind + 1]] - romans[s[ind]];
                ind += 2;
            } else {
                integerValue += romans[s[ind]];
                ind += 1;
            }
        }
        return integerValue;
    }
}
